server/onerror.ts

```diff
@@ -0,0 +1,193 @@
+    // When dealing with cross-globals a normal `instanceof` check doesn't work properly.
+    // See https://github.com/koajs/koa/issues/1466
+    // We can probably remove it once jest fixes https://github.com/facebook/jest/issues/2549.
+    const isNativeError =
+      Object.prototype.toString.call(err) === "[object Error]" ||
+      err instanceof Error;
+
+    // wrap non-error object
+    if (!isNativeError) {
+      let errMsg = err;
+      if (typeof err === "object") {
+        try {
+          errMsg = JSON.stringify(err);
+          // eslint-disable-next-line no-empty
+        } catch (e) {}
+      }
+      const newError = InternalError(`non-error thrown: ${errMsg}`);
+      // err maybe an object, try to copy the name, message and stack to the new error instance
+      if (err) {
+        if (err.name) {
+          newError.name = err.name;
+        }
+        if (err.message) {
+          newError.message = err.message;
+        }
+        if (err.stack) {
+          newError.stack = err.stack;
+        }
+        if (err.status) {
+          newError.status = err.status;
+        }
+        if (err.headers) {
+          newError.headers = err.headers;
+        }
+      }
+      err = newError;
+    }
+
+    // ENOENT support
+    if (err.code === "ENOENT") {
+      err.status = 404;
+    }
+
+    if (typeof err.status !== "number" || !http.STATUS_CODES[err.status]) {
+      err.status = 500;
+    }
+
+    // Push only unknown 500 errors to sentry
+    if (err.status === 500) {
+      requestErrorHandler(err, this);
+    }
+
+    const headerSent = this.headerSent || !this.writable;
+    if (headerSent) {
+      err.headerSent = true;
+    }
+
+    // nothing we can do here other
+    // than delegate to the app-level
+    // handler and log.
+    if (headerSent) {
+      return;
+    }
+
+    this.status = err.status;
+
+    this.set(err.headers);
+    const type = this.accepts("json", "html") || "json";
+    if (type === "html") {
+      html.call(this, err, this);
+    } else {
+      json.call(this, err, this);
+    }
+    this.type = type;
+
+    if (type === "json") {
+      this.body = JSON.stringify(this.body);
+    }
+    this.res.end(this.body);
+  };
+
+  return app;
+}
+
+/**
+ * json error handler
+ * @param {Error} err
+ */
+
+function json(err: any, ctx: Context) {
+  ctx.status = err.status;
+  let message = err.message || err.name;
+  let error;
+
+  if (err instanceof ValidationError) {
+    // super basic form error handling
+    ctx.status = 400;
+
+    if (err.errors && err.errors[0]) {
+      message = `${err.errors[0].message} (${err.errors[0].path})`;
+    }
+  }
+
+  if (err instanceof EmptyResultError || /Not found/i.test(message)) {
+    message = "Resource not found";
+    ctx.status = 404;
+    error = "not_found";
+  }
+
+  if (/Authorization error/i.test(message)) {
+    ctx.status = 403;
+    error = "authorization_error";
+  }
+
+  if (ctx.status === 500) {
+    message = "Internal Server Error";
```

Comment:
```suggestion
    message = "Internal server error";
```

Reply:
Existing `errorHandling` middleware [uses caps](https://github.com/outline/outline/blob/b0b7c7d647d50ac9fdcb6ea04ffdfb3341967307/server/middlewares/errorHandling.ts#L35). Although unlikely, but hope it doesn't introduce regressions :crossed_fingers:

Reply:
Yep 👍